dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/requests/McpRequestId.java

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.mcp.internal.requests;
+
+import java.math.BigDecimal;
+import java.util.Objects;
+
+import jakarta.json.bind.annotation.JsonbTypeDeserializer;
+import jakarta.json.bind.annotation.JsonbTypeSerializer;
+
+/**
+ * Stores the id of an MCP Request, which can be represented as a String or Number.
+ */
+@JsonbTypeSerializer(McpRequestIdSerializer.class)
+@JsonbTypeDeserializer(McpRequestIdDeserializer.class)
+public final class McpRequestId {
```

Comment:
Please implement `toString()` - it'll be helpful for inspecting the value in the debugger and when it gets traced out.